docs: refer to the app as the Mend Renovate app (#23336)
renovatebot/renovate: docs/usage/getting-started/private-packages.md

The recommended approaches in order of preference are:

1. **Self-hosted hostRules**: Configure a hostRules entry in the bot's `config.js` with the `hostType`, `matchHost` and `token` specified

> 1. **Renovate App with private modules from npmjs.org**: Add an encrypted `npmToken` to your Renovate config // DRAFT: should I change this line?

Reviewer: ```suggestion
1. **The Mend Renovate App with private modules from npmjs.org**: Add an encrypted `npmToken` to your Renovate config
```

> 1. **Renovate App with a private registry**: Add an plaintext `npmrc` plus an encrypted `npmToken` in config // DRAFT: should I change this line?

Reviewer: ```suggestion
1. **The Mend Renovate App with a private registry**: Add an plaintext `npmrc` plus an encrypted `npmToken` in config
```
